cpp/src/join/join_common_utils.cuh

```diff
@@ -42,25 +42,27 @@ constexpr auto remap_sentinel_hash(H hash, S sentinel)
 }
 
 /**
- * @brief Device functor to create a pair of hash value and index for a given row.
+ * @brief Device functor to create a pair of {hash_value, row_index} for a given row.
```

Comment:
As I mentioned in my last review, now that these are being used outside of joins I would probably also prefer them to be moved to a separate detail header that isn't component-specific (i.e not in `join` directories). @PointKernel what do you think?

Reply:
> to be moved to a separate detail header that isn't component-specific

Agreed. Maybe `include/cudf/detail/cuco_helpers.cuh`

Reply:
This may be out of the scope of this PR, because these utility functions are used in many places in other join APIs (hash join + mixed join). If I extract them out, more join files will be affected (like 4-5 more files). Let keep this PR as-is, and I'll make a follow up PR for that.

Here is the issue for the follow up work: https://github.com/rapidsai/cudf/issues/11176

Reply:
My idea was to create the helper header and make it used by `contains` and `semi_anti_join` in this PR and do the cleanup for other files in a separate PR. 

> Let's keep this PR as-is, and I'll make a follow up PR for that.

Works for me.